src/server/frontend_wayland/layer_shell_v1.cpp

```diff
@@ -585,19 +585,20 @@ void mf::LayerSurfaceV1::get_popup(struct wl_resource* popup)
 
 void mf::LayerSurfaceV1::ack_configure(uint32_t serial)
 {
-    while (!inflight_configures.empty() && inflight_configures.front().first < serial)
+    // Stale ack
+    if(inflight_configure && inflight_configure->first > serial)
     {
-        inflight_configures.pop_front();
+        return;
     }
 
-    if (inflight_configures.empty() || inflight_configures.front().first != serial)
+    if (!inflight_configure || inflight_configure->first != serial)
     {
         BOOST_THROW_EXCEPTION(std::runtime_error(
             "Could not find acked configure with serial " + std::to_string(serial)));
     }
```

Comment:
OK, so you asked why there was a queue, and this error answers: we want to error if the serial doesn't match a pending configure.

But I think the existing logic is wrong anyway:

1.It assumes that serial never overflows; and,
2. It processes stale acks (that you do fix)

So I think that

1. The queue should stay; and,
2. The logic should be something like...

```c++
    auto const acked_event = std::find_if(
        std::begin(inflight_configures),
        std::end(inflight_configures),
        [serial](auto& p) { return p.serial == serial; });

    if (acked_event == std::end(inflight_configures))
    {
        BOOST_THROW_EXCEPTION(std::runtime_error(
            "Could not find acked configure with serial " + std::to_string(serial)));
    }

    auto const acked_configure_size = acked_event->second;

    inflight_configures.erase(std::begin(inflight_configures), ++acked_event);
    
    if (!inflight_configures.empty())
    {
        // If a stale configure has been acked, we do no more
        return;
    }
```